queue_and_stack/dll_stack.py

```diff
@@ -6,13 +6,20 @@ class Stack:
     def __init__(self):
         self.size = 0
         # Why is our DLL a good choice to store our elements?
-        # self.storage = ?
+        self.storage = DoublyLinkedList()
 
-    def push(self, value):
-        pass
+    def push(self, value): # add to head
+        self.storage.add_to_head(value)
+        self.size = self.storage.length
+
+    def pop(self): #remove from head ? not 100% sure
+        value  = None
+        if self.size > 0:
+            value = self.storage.remove_from_head()
+            self.size = self.storage.length
+        return value
 
-    def pop(self):
-        pass
 
     def len(self):
-        pass
+        self.size = self.storage.length
```

Comment:
It's nice that you are keeping track of the length so as to have better control on the data structure well done.
This can be better improved like so
```
def len(self):
        # pass
        # get number of items in stack or queue
        if self.size > 0:
            return self.size
        else:
            return 0
```

Reply:
Thankyou 👍